Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArrayFeatureProcessor.cpp

```diff
@@ -573,6 +585,8 @@ namespace AZ
 
                 if (textureArray.NumMaterials() == 0)
                 {
+                    // Need to "destroy" the DecalTextureArray since erase only free up the space to be reused
+                    m_textureArrayList[decalLocation.textureArrayIndex] = {};
```

Comment:
Should that be something that's fixed in the IndexableList class? The way erase doesn't actually erase at the moment seems to call for future bugs like this.